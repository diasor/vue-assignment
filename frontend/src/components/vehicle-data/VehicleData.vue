<template>
  <div class="vehicle-data">
    <div class="data-cell data-cell__speed">
      <label>Current Speed</label>
    </div>
    <div class="data-cell data-cell__speed-bar">
      <LocalProgressBar paramType="line" :paramValue="speed" />
    </div>
    <div class="data-cell data-cell__soc">
      <label>State Of Change</label>
    </div>
    <div class="data-cell data-cell__soc-bar">
      <LocalProgressBar paramType="line" :paramValue="soc" />
    </div>
    <div class="data-cell data-cell__energy">
      <label>Energy</label>
    </div>
    <div class="data-cell data-cell__odo">
      <label>Odometer</label>
    </div>
    <div class="data-cell data-cell__energy-data">
      <label>{{ energy }}</label>
    </div>
    <div class="data-cell data-cell__odo-data">
      <label>{{ odo }}</label>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import LocalProgressBar from '@/components/progress-bar/LocalProgressBar';
import { Component, Prop } from 'vue-property-decorator';

@Component({
  name: 'VehicleData',
  components: {
    LocalProgressBar
  }
})
export default class VehicleData extends Vue {
  @Prop({ required: true, default: 0 }) speed!: number;
  @Prop({ required: true, default: 0 }) soc!: number;
  @Prop({ required: true, default: 0 }) energy!: number;
  @Prop({ required: true, default: 0 }) odo!: number;

  socPercentage: number = this.soc === 0 ? 0 : this.soc / 100;
}
</script>

<style scoped lang="scss">
@import '@/assets/sass/variables';

.vehicle-data {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(6, minmax(auto, 35px));
  grid-column-gap: 0px;
  grid-row-gap: 0px;
  width: 100%;
  margin: 0;
  padding: 1rem 2rem;
  border: 2px solid $background-color;
  border-radius: 0.5rem;
  background-color: $background-color;
}

.data-cell {
  display: flex;
  justify-content: flex-start;
  padding: 0.2rem;

  & label {
    font-family: $font-family;
    font-size: $title-medium;
    font-weight: bold;
    padding: 0.9rem 0 0.3rem 0;
  }

  &__speed {
    grid-column: 1 / 3;
    grid-row: 1;
  }

  &__speed-bar {
    grid-column: 1 / 3;
    grid-row: 2;
    margin: 0;
  }

  &__soc {
    grid-column: 1 / 3;
    grid-row: 3;
  }

  &__soc-bar {
    grid-column: 1 /3;
    grid-row: 4;
  }

  &__energy {
    grid-column: 1;
    grid-row: 5;
  }

  &__odo {
    grid-column: 2;
    grid-row: 5;
  }

  &__energy-data {
    grid-column: 1;
    grid-row: 6;
  }

  &__odo-data {
    grid-column: 2;
    grid-row: 6;
  }
}
</style>
