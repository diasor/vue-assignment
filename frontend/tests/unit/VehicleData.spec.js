import { mount } from "@vue/test-utils";
import VehicleData from "@/components/vehicle-data/VehicleData";

let wrapper;

beforeEach(() => {
  wrapper = mount(VehicleData, {
    propsData: {
      speed: 100.3,
      soc: 45.6,
      energy: 56.2,
      odo: 88
    }
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe("VehicleData.vue rendering", () => {
  // mounts the main component
  it("renders component", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  // renders one container class and 8 data-cells
  it("has a vehicle-data class and 8 data-cell classes", () => {
    expect(wrapper.findAll(".vehicle-data").length).toBe(1);
    expect(wrapper.findAll(".data-cell").length).toBe(8);
  });

  // tests that the socPercentage is correctly calculated
  it("has initial socPercentage value of 0.456", () => {
    expect(wrapper.vm.socPercentage).toBe(0.456);
  });

  // shows energy and odometer simply on the screen
  it("shows the proper energy and odometer values", () => {
    expect(wrapper.text()).toMatch("56.2");
    expect(wrapper.text()).toMatch("88");
  });
});
