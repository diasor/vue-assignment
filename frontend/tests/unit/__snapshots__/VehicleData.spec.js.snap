// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleData.vue rendering renders component 1`] = `
<div
  class="vehicle-data"
>
  <div
    class="data-cell data-cell__speed"
  >
    <label>
      Current Speed
    </label>
  </div>
   
  <div
    class="data-cell data-cell__speed-bar"
  >
    <div
      class="progress-bar"
      data-v-8248d938=""
      id="progress-bar-line"
    >
      <svg
        id="line-progress"
        style="height: 30px; width: 350px;"
      >
        <line
          stroke="#C0C0C0"
          stroke-width="30"
          x1="0"
          x2="100%"
          y1="50%"
          y2="50%"
        />
        <line
          stroke="#696969"
          stroke-width="30px"
          x1="0"
          x2="100.3%"
          y1="50%"
          y2="50%"
        />
        <text
          style="display: inherit; fill: #FFFFFF; font-size: 12px; font-family: Montserrat; text-shadow: 1px 1px 1px #000000;"
          x="43%"
          y="61%"
        >
           100.3% 
        </text>
      </svg>
    </div>
  </div>
   
  <div
    class="data-cell data-cell__soc"
  >
    <label>
      State Of Change
    </label>
  </div>
   
  <div
    class="data-cell data-cell__soc-bar"
  >
    <div
      class="progress-bar"
      data-v-8248d938=""
      id="progress-bar-line"
    >
      <svg
        id="line-progress"
        style="height: 30px; width: 350px;"
      >
        <line
          stroke="#C0C0C0"
          stroke-width="30"
          x1="0"
          x2="100%"
          y1="50%"
          y2="50%"
        />
        <line
          stroke="#696969"
          stroke-width="30px"
          x1="0"
          x2="45.6%"
          y1="50%"
          y2="50%"
        />
        <text
          style="display: inherit; fill: #FFFFFF; font-size: 12px; font-family: Montserrat; text-shadow: 1px 1px 1px #000000;"
          x="43%"
          y="61%"
        >
           45.6% 
        </text>
      </svg>
    </div>
  </div>
   
  <div
    class="data-cell data-cell__energy"
  >
    <label>
      Energy
    </label>
  </div>
   
  <div
    class="data-cell data-cell__odo"
  >
    <label>
      Odometer
    </label>
  </div>
   
  <div
    class="data-cell data-cell__energy-data"
  >
    <label>
      56.2
    </label>
  </div>
   
  <div
    class="data-cell data-cell__odo-data"
  >
    <label>
      88
    </label>
  </div>
</div>
`;
